core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt

```diff
@@ -0,0 +1,173 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.utils
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.core.content.ContextCompat.getExternalFilesDirs
+import androidx.core.content.edit
+import io.reactivex.Flowable
+import io.reactivex.processors.PublishProcessor
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.instance
+import org.kiwix.kiwixmobile.core.NightModeConfig
+import org.kiwix.kiwixmobile.core.NightModeConfig.Mode.Companion.from
+import java.io.File
+import java.util.Locale
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Manager for the Default Shared Preferences of the application.
+ */
+@Singleton
+class SharedPreferenceUtil @Inject constructor(context: Context) {
+  private val sharedPreferences: SharedPreferences =
+    PreferenceManager.getDefaultSharedPreferences(context)
+  private val _prefStorages = PublishProcessor.create<String>()
+  val prefStorages = _prefStorages.startWith { prefStorage }
```

Comment:
on second thought can you change this to a `get() =`?

Reply:
Okay from now on, i'll be first thinking to use getters for properties like this and always for sharedPrefs as getters are much better than normal initializers. Got it!

Reply:
They are not better. It depends on the use case.

It would require some explanation of how Rx streams work but essentially without the getter it would have been possible to miss changes to prefStorages which is not desirable.

Only use `get() =` when you want the RHS to run every time. Sometimes you don't for example

```
data class Person(val age:Int){
val canPurchaseAlcohol = age >= 18
}
```

we wouldn't want to run the RHS every time we query `canPurchaseAlcohol`, it would be wasteful. Granted in this example a persons age can never change

Reply:
I think i got what you just told, but why did you use a dataClass for consuming just one single parameter in the constructor, you could've used a normal class as well, but why data class?(Because of its auto-generated getters and setters?)

Reply:
I could have used a regular class but if a class holds data then my finger just plop a `data` there.
Getters and setter are not contributed by it being a data class, that is an intrinsic property of `val/var`

![image](https://user-images.githubusercontent.com/3358282/90614467-9a043380-e202-11ea-8c06-89e52f573c4b.png)

Reply:
>  Getters and setter are not contributed by it being a data class, that is an intrinsic property of val/var

My whole life was a scam! I was told that data classes generate getters and setters :cry: